models/zeroex/arbitrum/zeroex_arbitrum_native_fills.sql

```diff
@@ -0,0 +1,240 @@
+      LEFT JOIN prices.usd tp ON
+          date_trunc('minute', evt_block_time) = tp.minute 
+          AND CASE
+                  -- set native token to wrapped version
+                    WHEN fills.takerToken = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN '0x82af49447d8a07e3bd95bd0d56f35241523fbab1'
+                    ELSE fills.takerToken
+              END = tp.contract_address
+      LEFT JOIN prices.usd mp ON
+          DATE_TRUNC('minute', evt_block_time) = mp.minute  
+          AND CASE
+                  -- set native token to wrapped version
+                    WHEN fills.makerToken = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' THEN '0x82af49447d8a07e3bd95bd0d56f35241523fbab1'
+                    ELSE fills.makerToken
+              END = mp.contract_address
+      LEFT OUTER JOIN {{ ref('tokens_erc20') }} mt ON mt.contract_address = fills.makerToken
+      LEFT OUTER JOIN {{ ref('tokens_erc20') }} tt ON tt.contract_address = fills.takerToken
+       where 1=1  and mp.blockchain = 'arbitrum' and tp.blockchain = 'arbitrum'  
+                {% if is_incremental() %}
+                AND evt_block_time >= date_trunc('day', now() - interval '1 week')
+                {% endif %}
+                {% if not is_incremental() %}
+                AND evt_block_time >= '{{zeroex_v3_start_date}}'
+                {% endif %}
+
+    ),
+
+    all_fills as (
+    
+   
+
+    SELECT * FROM v4_limit_fills
+
+    UNION ALL
+
+    SELECT * FROM v4_rfq_fills
+
+    UNION ALL
+    
+    SELECT * FROM otc_fills
+    )
+            SELECT distinct 
+                all_fills.block_time as block_time,
+                protocol_version as version,
+                date_trunc('day', all_fills.block_time) as block_date,
+                transaction_hash as tx_hash,
+                evt_index,
+                maker_address as maker,
+                taker_address as taker,
+                maker_token,
+                maker_token_filled_amount_raw as maker_token_amount_raw,
+                taker_token_filled_amount_raw as taker_token_amount_raw,
+                maker_symbol,
+                CASE WHEN lower(ts.symbol) > lower(ms.symbol) THEN concat(ms.symbol, '-', ts.symbol) ELSE concat(ts.symbol, '-', ms.symbol) END AS token_pair,
+                CAST(ARRAY() as array<bigint>) as trace_address,
+                maker_asset_filled_amount maker_token_amount,
+                taker_token,
+                taker_symbol,
+                taker_asset_filled_amount taker_token_amount,
+                fee_recipient_address,
+                volume_usd,
+                cast(protocol_fee_paid_eth as double),
+                'arbitrum' as blockchain,
+                all_fills.contract_address,
+                native_order_type,
+                tx.from AS tx_from,
+                tx.to AS tx_to
+            FROM all_fills
+            INNER JOIN {{ source('arbitrum', 'transactions')}} tx ON all_fills.transaction_hash = tx.hash
```

Comment:
add incremental/non-incremental date filters on base table in join conditions please. adding `block_number` may be beneficial as well